velox/functions/lib/tests/Re2FunctionsTest.cpp

```diff
@@ -384,6 +402,161 @@ TEST_F(Re2FunctionsTest, likePattern) {
   EXPECT_EQ(like("abc", "MEDIUM POLISHED%"), false);
 }
 
+TEST_F(Re2FunctionsTest, likeDeterminePatternKind) {
+  auto testPattern =
+      [&](StringView pattern, PatternKind patternKind, vector_size_t length) {
+        EXPECT_EQ(
+            determinePatternKind(pattern), std::make_pair(patternKind, length));
+      };
+
+  testPattern("_", PatternKind::kExactlyN, 1);
+  testPattern("____", PatternKind::kExactlyN, 4);
+  testPattern("%", PatternKind::kAtLeastN, 0);
+  testPattern("__%%__", PatternKind::kAtLeastN, 4);
+  testPattern("%_%%", PatternKind::kAtLeastN, 1);
+  testPattern("_b%%__", PatternKind::kGeneric, 0);
+  testPattern("%_%p", PatternKind::kGeneric, 0);
+
+  testPattern("presto", PatternKind::kFixed, 6);
+  testPattern("hello", PatternKind::kFixed, 5);
+  testPattern("a", PatternKind::kFixed, 1);
+  testPattern("helloPrestoWorld", PatternKind::kFixed, 16);
+  testPattern("aBcD_", PatternKind::kGeneric, 0);
+  testPattern("%aBc_D%", PatternKind::kGeneric, 0);
+
+  testPattern("presto%", PatternKind::kPrefix, 6);
+  testPattern("hello%%", PatternKind::kPrefix, 5);
+  testPattern("a%", PatternKind::kPrefix, 1);
+  testPattern("helloPrestoWorld%%%", PatternKind::kPrefix, 16);
+  testPattern("aBcD%%e%", PatternKind::kGeneric, 0);
+  testPattern("aBc_D%%", PatternKind::kGeneric, 0);
+
+  testPattern("%presto", PatternKind::kSuffix, 6);
+  testPattern("%%hello", PatternKind::kSuffix, 5);
+  testPattern("%a", PatternKind::kSuffix, 1);
+  testPattern("%%%helloPrestoWorld", PatternKind::kSuffix, 16);
+  testPattern("%%_%aBcD", PatternKind::kGeneric, 0);
+  testPattern("%%a%%BcD", PatternKind::kGeneric, 0);
+  testPattern("foo%bar", PatternKind::kGeneric, 0);
+}
+
+TEST_F(Re2FunctionsTest, likePatternWildcard) {
+  auto like = [&](std::optional<std::string> str, std::string pattern) {
```

Comment:
Can you remove std::optional here and pass std::make_optional(str) to evaluateOnce?

Reply:
Thanks for the suggestion!